### ISSUE TITLE
Token Expiration overspecified --  Erik Kline

### ISSUE THREAD

**Issue by hannestschofenig** [ace-wg/ace-oauth#190]
5.10.3.  Token Expiration

   Depending on the capabilities of the RS, there are various ways in
   which it can verify the expiration of a received access token.  Here
   follows a list of the possibilities including what functionality they
   require of the RS.

   o  The token is a CWT and includes an "exp" claim and possibly the
      "nbf" claim.  The RS verifies these by comparing them to values
      from its internal clock as defined in [RFC7519].  In this case the
      RS's internal clock must reflect the current date and time, or at
      least be synchronized with the AS's clock.  How this clock
      synchronization would be performed is out of scope for this
      specification.

   o  The RS verifies the validity of the token by performing an
      introspection request as specified in Section 5.9.  This requires
      the RS to have a reliable network connection to the AS and to be
      able to handle two secure sessions in parallel (C to RS and RS to
      AS).

   o  In order to support token expiration for devices that have no
      reliable way of synchronizing their internal clocks, this
      specification defines the following approach: The claim "exi"
      ("expires in") can be used, to provide the RS with the lifetime of
      the token in seconds from the time the RS first receives the
      token.  This mechanism only works for self-contained tokens, i.e.
      CWTs and JWTs.  For CWTs this parameter is encoded as unsigned
      integer, while JWTs encode this as JSON number.

   o  Processing this claim requires that the RS does the following:

      *  For each token the RS receives, that contains an "exi" claim:
         Keep track of the time it received that token and revisit that
         list regularly to expunge expired tokens.

      *  Keep track of the identifiers of tokens containing the "exi"
         claim that have expired (in order to avoid accepting them
         again).  In order to avoid an unbounded memory usage growth,
         this MUST be implemented in the following way when the "exi"
         claim is used:

         +  When creating the token, the AS MUST add a 'cti' claim ( or
            'jti' for JWTs) to the access token.  The value of this
            claim MUST be created as the binary representation of the
            concatenation of the identifier of the RS with a sequence
            number counting the tokens containing an 'exi' claim, issued
            by this AS for the RS.

         +  The RS MUST store the highest sequence number of an expired
            token containing the "exi" claim that it has seen, and treat
            tokens with lower sequence numbers as expired.

   If a token that authorizes a long running request such as a CoAP
   Observe [RFC7641] expires, the RS MUST send an error response with
   the response code equivalent to the CoAP code 4.01 (Unauthorized) to
   the client and then terminate processing the long running request.

**Comment by LudwigSeitz** [comment]
I'm not seeing an issue description here. Or is the issue that this text is too much detail?

**Comment by LudwigSeitz** [comment]
Since I don't have more information on this I'm closing it.
